app/ts/context/BalanceContext.tsx

```diff
@@ -0,0 +1,94 @@
+import { Signal, useSignal } from '@preact/signals'
+import { BigNumber, ethers } from 'ethers'
+import { ComponentChildren, createContext } from 'preact'
+import { useContext } from 'preact/hooks'
+import { useAsyncState } from '../library/preact-utilities.js'
+import { assertsExternalProvider, assertUnreachable, isEthereumObservable } from '../library/utilities.js'
+
+type Balance =
+	| {
+			state: 'initial'
+			checkBalance: () => void
+	  }
+	| {
+			state: 'updating'
+			reset: () => void
+	  }
+	| {
+			state: 'updated'
+			balance: BigNumber
+			updated_at: Date
+	  }
+	| {
+			state: 'failed'
+			error: Error
+			reset: () => void
+	  }
+
+export type BalanceStore = Signal<Balance>
+export function createBalanceStore() {
+	const { value: query, waitFor, reset } = useAsyncState<BigNumber>()
+
+	const checkBalance = () => {
+		waitFor(async () => {
+			assertsExternalProvider(window.ethereum)
+			const provider = new ethers.providers.Web3Provider(window.ethereum)
+			const signer = provider.getSigner()
+			const address = await signer.getAddress()
```

Comment:
I think this address should be stored elsewhere, as when you are showing the address, this balance can get out of sync when user switches address, as this retrieves the balance of signers address, and not the one showed in the application

Reply:
The address is now shared and should be synced with address set during account connection and account change events.